core/src/main/java/com/linecorp/armeria/client/Clients.java

```diff
@@ -330,11 +327,7 @@ public static SafeCloseable withHttpHeader(AsciiString name, String value) {
      */
     public static SafeCloseable withHttpHeaders(Function<HttpHeaders, HttpHeaders> headerManipulator) {
         requireNonNull(headerManipulator, "headerManipulator");
-        return withContextCustomizer(ctx -> {
-            final Attribute<HttpHeaders> attr = ctx.attr(HTTP_HEADERS);
-            final HttpHeaders headers = attr.get();
-            attr.set(headerManipulator.apply(headers != null ? headers : new DefaultHttpHeaders()));
-        });
+        return withContextCustomizer(ctx -> headerManipulator.apply(ctx.additionalRequestHeaders()));
```

Comment:
Just curious. Do we need to change `Function<HttpHeaders, HttpHeaders> headerManipulator` to `Consumer<HttpHeaders>`? It seems that we cannot manipulate the headers if the `headerManipulator` returns a new `HttpHeaders`.

Reply:
I think we can manipulate the `HttpHeaders` regardless of returning new `HttpHeaders` or not, because the `headerManipulator`'s role is manipulating the `HttpHeaders` which it takes as an argument.
I think we can change this parameter type to `Consumer<>` but all of the `add()` and `set()` methods in `HttpHeaders` interface return the `HttpHeaders` itself So we can use the same policy here as well. WDYT?